### PR TITLE
[BE] Enable Ruff's Flake8 PYI034

### PR DESCRIPTION
**Pull request by isdanni**
Enable [non-self-return-type (PYI034)](https://docs.astral.sh/ruff/rules/non-self-return-type/#non-self-return-type-pyi034)

Link: #110950

**EDIT**: to newly added reviewers, please ignore the request, it's due to a rebase error :sweat_smile: 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torchgen/utils.py [hunk 1]
```diff
@@ -486,7 +488,7 @@ def union(*args: "OrderedSet[T]") -> "OrderedSet[T]":
     def __or__(self, other: "OrderedSet[T]") -> "OrderedSet[T]":
```
Comment (Skylion007) [inline torchgen/utils.py:488]: Shouldn't more of these also be typing_extensions Self?

File: torch/nn/modules/container.py [hunk 2]
```diff
@@ -626,7 +627,7 @@ def __len__(self) -> int:
     def __iter__(self) -> Iterator[Any]:
         return iter(self[i] for i in range(len(self)))
 
-    def __iadd__(self, parameters: Iterable[Any]) -> 'ParameterList':
```
[the remark this answers is not in the document]
Comment (XuehaiPan) [reply torch/nn/modules/container.py:629]: Should also update the type hint for `extend()`:

```diff
- def extend(self, values: Iterable[Any]) -> 'ParameterList':
+ def extend(self, values: Iterable[Any]) -> Self:
```

It is a false negative for `extend()`.

File: torch/nn/modules/container.py [hunk 3]
```diff
@@ -314,7 +315,7 @@ def __len__(self) -> int:
     def __iter__(self) -> Iterator[Module]:
         return iter(self._modules.values())
 
-    def __iadd__(self, modules: Iterable[Module]) -> 'ModuleList':
+    def __iadd__(self, modules: Iterable[Module]) -> Self:
```
Comment (kit1980) [inline torch/nn/modules/container.py:318]: Another wrong one?

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**